polaris-react/src/types.ts

```diff
@@ -206,9 +206,11 @@ export interface ActionListItemDescriptor
 
 export interface ActionListSection {
   /** Section title */
-  title?: string;
+  title?: React.ReactNode;
```

Comment:
Documentation wise, would it be better to use `string | ReactNode` so it's understand only a string is provided?
But on that note having the word `title` implies its a string in most cases so perhaps we can provide a section `header` or `headerContent` prop which allows for content that isn't necessarily a title.

Reply:
> But on that note having the word title implies its a string in most cases so perhaps we can provide a section header or headerContent prop which allows for content that isn't necessarily a title

`title` is being used as a `React.ReactNode` in a [few places](https://grokt.shopify.io/results?q=%22title%3F%3A+React.ReactNode%22+r%3Apolaris) already but I also think we might introduce an extra prop that won't get used as much.

Reply:
Yup I'm aware it's used elsewhere; I just figured the naming felt wrong, but I can live with it since we want to minimize the changes.

Reply:
Yeah, I agree. I thought about renaming the `title` but that seems like a major update for a small semantic issue. 

I'll add the `string` type to it.